public/app/plugins/datasource/loki/queryUtils.ts

```diff
@@ -76,6 +77,23 @@ export function getHighlighterExpressionsFromQuery(input = ''): string[] {
   return results;
 }
 
+/**
+ * Returns search terms from a LogQL query with variables.
+ * Given a variable name $search with value foo:
+ * E.g., `{} |= $search |= "bar" != "baz"` returns `['foo']`.
```

Comment:
```suggestion
 * E.g., `{} |= $search |= "bar" != "baz"` returns `['foo', 'bar']`.
```

Right?

Reply:
No. This function works in combination with the existing one, which combines both values coming from variables and those identified in the query.

Reply:
That's why it should return the value of `$search` and the value of `bar`, imo.